UnityProject/Assets/Scripts/Input System/MouseInputController.cs

```diff
@@ -306,9 +320,67 @@ private void CheckHover()
 			}
 
 			hit.transform.SendMessageUpwards("OnHover", SendMessageOptions.DontRequireReceiver);
+
+			// check if interaction is enabled
+			if (!UIManager.IsMouseInteractionDisabled)
+			{
+				// check if item in main hand is Cable
+				if (CheckIfCableInHand(out CableCoil cable))
+				{
+					// get mouse position
+					Vector3 mousePosition = Camera.main.ScreenToWorldPoint(CommonInput.mousePosition);
+					// round mouse position
+					Vector3Int roundedMousePosition = Vector3Int.RoundToInt(mousePosition);
+
+					if (roundedMousePosition != lastMouseWordlPositionInt)
+					{
+						lastMouseWordlPositionInt = roundedMousePosition;
+
+						// get metaTileMap and top tile
+						MetaTileMap metaTileMap = MatrixManager.AtPoint(roundedMousePosition, false).MetaTileMap;
+						LayerTile topTile = metaTileMap.GetTile(metaTileMap.WorldToCell(mousePosition), true);
+
+						if (topTile && (topTile.LayerType == LayerType.Base || topTile.LayerType == LayerType.Underfloor))
+						{
+							// move cable placement visualisation to rounded mouse position and enable it
+							cablePlacementVisualisation.transform.position = roundedMousePosition - new Vector3(0.5f, 0.5f, 0);
+							cablePlacementVisualisation.gameObject.SetActive(true);
+						}
+						// disable visualisation if active
+						else if (cablePlacementVisualisation.gameObject.activeSelf)
+							cablePlacementVisualisation.gameObject.SetActive(false);
+					}
+				}
+				// disable visualisation if item in hand is not cable and cablePlacementVisualisation is active
+				else if (cablePlacementVisualisation.gameObject.activeSelf)
+					cablePlacementVisualisation.gameObject.SetActive(false);
+			}
 		}
 	}
 
+	/// <summary>
+	/// Check if item in main hand has CableCoil component
+	/// </summary>
+	/// <param name="cable">CableCoil component</param>
+	/// <returns>true if object has component</returns>
+	private bool CheckIfCableInHand(out CableCoil cable)
```

Comment:
There must be a better way than having a whole method to just check if a cable is in hand here?

Reply:
Could use the cable itemtrait?

Reply:
For now I've changed it to `if(UIManager.Hands.CurrentSlot.ItemObject && UIManager.Hands.CurrentSlot.ItemObject.TryGetComponent(out CableCoil cable))`
@ThatDan123 what do you mean by 'itemtrait'?

Reply:
he means something like `Validations.HasItemTrait(UIManager.Hands.CurrentSlot.ItemObject, CommonTraits.Instance.Cable)`

Reply:
Your method will work just as fine, its just an alternative to look up instead of trying to get the component:

Validations.HasItemTrait(object, itemtrait)

In your case would be:

If(Validations.HasItemTrait(UIManager.Hands.CurrentSlot.ItemObject, CommonTraits.Instance.Cable))

Reply:
damn it foma :p